src/js/components/history/HistoryItem.tsx

```diff
@@ -0,0 +1,192 @@
+
+  constructor(props) {
+    super(props);
+  }
+
+  private width = 100; // these should match baseSignals in demonstrations.ts
+  private height = 100; //
+
+  public handleClick = (historyId: number) => {
+
+  }
+  public handleDragStart = (historyId: number) => {
+    this.props.startDragging(HistoryDraggingState({historyId}));
+
+    sg.set(MODE, 'channels');
+    ctrl.update();
+  }
+
+  public handleDragEnd = (evt: React.DragEvent<HTMLDivElement>, opts?) => {
+    const props = this.props;
+    const sel = sg.get(SELECTED);
+    const cell = sg.get(CELL);
+    const dropped = tupleid(sel) && tupleid(cell);
+
+    try {
+      const lyraId = +sel.mark.role.split('lyra_')[1]; // id of thing that was dropped onto
+
+      if (dropped) {
+        const channel = channelName(cell.key);
+        let fieldName, dsId;
+        if (channel === 'x' || channel === 'y' || channel === 'color' || channel === 'size') {
+          // set scale
+          let channelScaleIds = this.props.history.getIn(["guides"])
+            .map((g) => {
+              return channel == 'x' || channel == 'y' ? g.scale : channel == 'size' ? g[channel] : g.fill;
+            })
+            .filter((scaleId) => {
+              let scaleName = scaleId ? this.props.history.getIn(["scales"]).get(scaleId).name : null;
+              return scaleName === channel;
+            });
+
+          fieldName = channelScaleIds.map((scaleId) => {
+            let scaleRecord = this.props.history.getIn(["scales"]).get(scaleId);
+            return scaleRecord.get('_domain').length > 0 ?  scaleRecord.get('_domain')[0].field : null;
+          }).first();
+
+          dsId = channelScaleIds.map((scaleId) => {
+            let scaleRecord = this.props.history.getIn(["scales"]).get(scaleId);
+            return scaleRecord.get('_domain').length > 0 ?  scaleRecord.get('_domain')[0].data : null;
+          }).first();
+
+
+        }
+
+        const bindField = this.props.history.getIn(["datasets", String(dsId), "_schema", fieldName]).toJS();
+        vega.extend(bindField, opts); // Aggregate or Bin passed in opts.
+        props.bindChannel(dsId, bindField, lyraId, cell.key);
+      } else {
+        // update the whole symbol/mark: shape, size
+        // the history has the signals
+        let relevantProps = ["size", "shape", "fill", "fillOpacity", "stroke",  "strokeWidth"];
+        let historyMark = this.props.history.getIn(["marks", String(lyraId)]); // assume mark is the same as before. wanna revert to old settings of old mark
+
+        batchGroupBy.start();
+        relevantProps.forEach((prop) => {
+          let historyProp = historyMark.getIn(["encode", "update", prop]);
+          if (historyProp.signal) {
+            let historySigVal = this.props.history.getIn(["signals", historyProp.signal]).value;
+            this.props.setSignal(historySigVal, historyProp.signal); // update Store
+            sg.set(historyProp.signal, historySigVal, false); // update Vega
```

Comment:
i suggest doing
```
sg.set(historyProp.signal, historySigVal, false);
ctrl.update();
```
instead of dispatching `setSignal`

Reply:
unless there's something that doesn't work without `setSignal`. i'm suggesting this because this is what the inspectors do to update these types of values

Reply:
didn't need setSignal!